server/container_create_linux.go

```diff
@@ -1075,7 +1076,7 @@ func (s *Server) addOCIBindMounts(ctx context.Context, ctr ctrfactory.Container,
 			return nil, nil, errors.New("mount.ContainerPath is empty")
 		}
 		if m.Image != nil && m.Image.Image != "" {
-			mountPoint, imageID, err := s.mountImage(ctx, m.Image.Image, mountLabel)
+			mountPoint, imageID, err := s.mountImage(ctx, specgen, m.Image.Image, m.ContainerPath)
 			if err != nil {
 				return nil, nil, fmt.Errorf("mount image: %w", err)
 			}
```

Comment:
We would need to `continue` the loop here because we don't have a local mount point any more. Not sure if that's the overall right approach because it would make the implementation clumsy from my point of view. :thinking: